### DISCUSSION TITLE
Non-euclidean Voronoi

### DISCUSSION THREAD

**Discussion by micycle1** [gwlucastrig/Tinfour#87] [Q&A]
Could the Tinfour Voronoi implementation easily generalise to accept other distance metrics (manhattan for instance)? Or is the Euclidean nature very much tied up in with the algorithm?

Also relevant -- Weighted Voronoi Diagrams.

**Comment by gwlucastrig** [comment]
Well, I was afraid somebody might ask that question one of these days...   Tinfour's Voronoi  implementation is constructed on top of the Delaunay. And its Delaunay is deeply rooted in coordinate systems with a Euclidean metric.

Really, the Voronoi in Tinfour was intended mostly as a way of rendering example diagrams in articles about the Natural Neighbor Interpolator. I tried to provide enough API elements that it could be useful for data analysis and so forth, but it has limited capabilities.

I find the whole problem of using other distance metrics really, really interesting. Someday, I'd like to take it on. But right now, I'm afraid that's out of my reach.

Gary
